example/bench/bench.cpp

```diff
@@ -35,21 +35,21 @@ int main(int argc, char** argv) {
     try {
         arb::execution_context context;
 #ifdef ARB_MPI_ENABLED
-        aux::with_mpi guard(&argc, &argv);
-        context.distributed = mpi_context(MPI_COMM_WORLD);
+        with_mpi guard(argc, argv, false);
+        context.distributed = std::make_shared<arb::distributed_context>(arb::mpi_context(MPI_COMM_WORLD));
```

Comment:
Now that we're using a `shared_ptr` for the distributed context, we can have context-makers like `arb::mpi_context` return a `shared_ptr` directly.

This points towards moving away from value-like semantics for the type-erasing `distributed_context` class, but that's something I think I will have to arm wrestle @bcumming over, and can be the topic of a later issue.